sdks/python/apache_beam/runners/common.py

```diff
@@ -761,6 +761,17 @@ def __init__(self,
       self.current_window_index = None
       self.stop_window_index = None
 
+    # TODO(https://github.com/apache/beam/issues/28776): Remove caching after
+    # fully rolling out.
+    # If true, always recalculate window args. If false, has_cached_window_args
+    # and has_cached_window_batch_args will be set to true if the corresponding
+    # self.args_for_process,have been updated and should be reused directly.
+    self.recalculate_window_args = (
+        self.has_windowed_inputs or 'disable_global_windowed_args_caching' in
+        RuntimeValueProvider.experiments)
+    self.has_cached_window_args = False
```

Comment:
I think `has_cached_args` or `has_cached_windowed_args` might be a better name .

Reply:
Sorry, I forgot to push this after I was running the test and already merged.